core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1098,7 +1085,6 @@ class DAGScheduler(
           logInfo(s"Marking $failedStage (${failedStage.name}) as failed " +
             s"due to a fetch failure from $mapStage (${mapStage.name})")
           markStageAsFinished(failedStage, Some(failureMessage))
-          runningStages -= failedStage
```

Comment:
This was redundant, since `markStagesAsFinished` removes the stage from `runningStages`.